### ISSUE TITLE
v21.5.0 would not work with custom options

### ISSUE THREAD

**Issue by AngusFu** [jdalrymple/gitbeaker#894]
**Description**

SEE https://github.com/jdalrymple/gitbeaker/commit/8b549da700d5a91ab34487f463b169bea37970aa#r39933403

**Steps to reproduce**

```js
const { Projects } = require('@gitbeaker/node');

const api = new Projects({
  host: 'https://gitlab.XXXXXXX.COM',
  token: 'SOME_TOKEN'
});

console.log(api.url);  // v21.5.0 would still print out "https://gitlab.com/api/v4/"
```

**Possible fixes**

Just fix the `extendClass` function in [packages/gitbeaker-requester-utils/src/RequesterUtils.ts](https://github.com/jdalrymple/gitbeaker/blob/a32e9db9b20731f81643769e1742a30a3cb35e88/packages/gitbeaker-requester-utils/src/RequesterUtils.ts#L81) 

**Why**

```ts
function extendClass<T extends Constructable>(Base: T, customConfig: object): T {Justin Dalrymple, 4 days ago: • Remove Gitlab Instance dependency for CLI tes…
  return class extends Base {
    constructor(...options: any[]) {
      super({ ...options, ...customConfig });
    }
  };
}
```

![image](https://user-images.githubusercontent.com/8180186/84737560-bd301e00-afda-11ea-98c9-2955879bd44b.png)

Another demo:

```js
function extendClass(Base, customConfig) {
  return class extends Base {
    constructor(...options) {
      super({ ...options, ...customConfig });
    }
  };
}

const Klass = extendClass(
  class Base {
    constructor(config) {
      console.log(config);
      this.host = config.host || 'http://xxxx.yyyy'
    }
  },
  { time: Date.now() }
)

console.log( new Klass({ host: 'https://my.gitlab.com' }) )
```

**Comment by ChhXin** [comment]
:frowning:

**Comment by DerPate** [comment]
Can confirm the same Problem here.

**Comment by jdalrymple** [comment]
I'll give it a look! Thanks for the well formed bug report :heart_eyes:

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`21.7.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/21.7.0) :rocket:
